File: server/internal/planner/model/trip.go
package model

import (
	"time"

	"gorm.io/gorm"
)

// Trip 行程模型
type Trip struct {
	ID          uint           `json:"id" gorm:"primaryKey;autoIncrement"`
	Title       string         `json:"title" gorm:"type:varchar(200);not null;comment:行程标题"`
	Description string         `json:"description" gorm:"type:text;comment:行程描述"`
	Destination string         `json:"destination" gorm:"type:varchar(200);comment:目的地"`
	StartDate   *time.Time     `json:"start_date" gorm:"type:timestamp;comment:开始日期"`
	EndDate     *time.Time     `json:"end_date" gorm:"type:timestamp;comment:结束日期"`
	CoverImage  string         `json:"cover_image" gorm:"type:varchar(500);comment:封面图片URL"`
	Status      string         `json:"status" gorm:"type:varchar(20);default:draft;comment:状态(draft/planning/confirmed/completed/cancelled)"`
	CreatorID   uint           `json:"creator_id" gorm:"not null;index;comment:创建者ID"`
	Budget      float64        `json:"budget" gorm:"type:decimal(10,2);default:0;comment:预算"`
	Activities  []Activity     `json:"activities" gorm:"foreignKey:TripID"`
	Members     []TripMember   `json:"members" gorm:"foreignKey:TripID"`
	CreatedAt   time.Time      `json:"created_at"`
	UpdatedAt   time.Time      `json:"updated_at"`
	DeletedAt   gorm.DeletedAt `json:"-" gorm:"index"`
}

// TableName 指定表名
func (Trip) TableName() string {
	return "trips"
}

// TripMember 行程成员模型
type TripMember struct {
	ID        uint      `json:"id" gorm:"primaryKey;autoIncrement"`
	TripID    uint      `json:"trip_id" gorm:"not null;index;comment:行程ID"`
	UserID    uint      `json:"user_id" gorm:"not null;index;comment:用户ID"`
	Role      string    `json:"role" gorm:"type:varchar(20);default:member;comment:角色(owner/editor/viewer)"`
	JoinedAt  time.Time `json:"joined_at"`
	CreatedAt time.Time `json:"created_at"`
}

// TableName 指定表名
func (TripMember) TableName() string {
	return "trip_members"
}

File: ai-engine/app/models/trip.py
"""
行程模型

定义行程相关的数据结构。
"""

from datetime import date, datetime
from typing import Optional

from pydantic import BaseModel, Field


class Trip(BaseModel):
    """行程模型"""
    trip_id: str = Field(..., description="行程唯一标识")
    user_id: str = Field(..., description="用户ID")
    title: str = Field(..., description="行程标题")
    destination: str = Field(..., description="目的地")
    start_date: date = Field(..., description="开始日期")
    end_date: date = Field(..., description="结束日期")
    days: int = Field(..., ge=1, description="行程天数")
    travelers_count: int = Field(1, ge=1, description="出行人数")
    total_budget: Optional[float] = Field(None, description="总预算(元)")
    total_cost: Optional[float] = Field(None, description="实际总费用(元)")
    status: str = Field("draft", description="行程状态(draft/confirmed/completed/cancelled)")
    cover_image: Optional[str] = Field(None, description="封面图片URL")
    tags: list[str] = Field(default_factory=list, description="标签")
    notes: Optional[str] = Field(None, description="备注")
    created_at: datetime = Field(default_factory=datetime.now, description="创建时间")
    updated_at: datetime = Field(default_factory=datetime.now, description="更新时间")


class TripSummary(BaseModel):
    """行程摘要"""
    trip_id: str = Field(..., description="行程ID")
    title: str = Field(..., description="行程标题")
    destination: str = Field(..., description="目的地")
    start_date: date = Field(..., description="开始日期")
    end_date: date = Field(..., description="结束日期")
    days: int = Field(..., description="行程天数")
    status: str = Field(..., description="行程状态")
    cover_image: Optional[str] = Field(None, description="封面图片")
    tags: list[str] = Field(default_factory=list, description="标签")
